### PR TITLE
fix: event import

### LINKED ISSUE #4613: Event import end point not working
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-server

**Current behavior:**

Event import end point gives the following error
```python
2018-01-31T14:49:27.762635+00:00 app[web.1]: get_file_from_request() INVOKED. We have: request.files = ImmutableMultiDict([])
2018-01-31T14:49:27.765722+00:00 app[web.1]: [2018-01-31 14:49:27 +0000] [54] [ERROR] Exception in worker process
2018-01-31T14:49:27.765724+00:00 app[web.1]: Traceback (most recent call last):
2018-01-31T14:49:27.765726+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/arbiter.py", line 578, in spawn_worker
2018-01-31T14:49:27.765727+00:00 app[web.1]:     worker.init_process()
2018-01-31T14:49:27.765728+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/base.py", line 131, in init_process
2018-01-31T14:49:27.765730+00:00 app[web.1]:     self.run()
2018-01-31T14:49:27.765731+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 124, in run
2018-01-31T14:49:27.765732+00:00 app[web.1]:     self.run_for_one(timeout)
2018-01-31T14:49:27.765734+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 68, in run_for_one
2018-01-31T14:49:27.765735+00:00 app[web.1]:     self.accept(listener)
2018-01-31T14:49:27.765736+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 30, in accept
2018-01-31T14:49:27.765737+00:00 app[web.1]:     self.handle(listener, client, addr)
2018-01-31T14:49:27.765739+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 135, in handle
2018-01-31T14:49:27.765740+00:00 app[web.1]:     self.handle_request(listener, req, client, addr)
2018-01-31T14:49:27.765741+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 176, in handle_request
2018-01-31T14:49:27.765742+00:00 app[web.1]:     respiter = self.wsgi(environ, resp.start_response)
2018-01-31T14:49:27.765743+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1997, in __call__
2018-01-31T14:49:27.765745+00:00 app[web.1]:     return self.wsgi_app(environ, start_response)
2018-01-31T14:49:27.765746+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/raven/middleware.py", line 100, in __call__
2018-01-31T14:49:27.765747+00:00 app[web.1]:     iterable = self.application(environ, start_response)
2018-01-31T14:49:27.765748+00:00 app[web.1]:   File "/app/app/__init__.py", line 71, in __call__
2018-01-31T14:49:27.765749+00:00 app[web.1]:     return self.app(environ, start_response)
2018-01-31T14:49:27.765750+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
2018-01-31T14:49:27.765751+00:00 app[web.1]:     response = self.full_dispatch_request()
2018-01-31T14:49:27.765753+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
2018-01-31T14:49:27.765754+00:00 app[web.1]:     rv = self.dispatch_request()
2018-01-31T14:49:27.765755+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
2018-01-31T14:49:27.765756+00:00 app[web.1]:     return self.view_functions[rule.endpoint](**req.view_args)
2018-01-31T14:49:27.765757+00:00 app[web.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/flask_jwt/__init__.py", line 177, in decorator
2018-01-31T14:49:27.765758+00:00 app[web.1]:     return fn(*args, **kwargs)
2018-01-31T14:49:27.765760+00:00 app[web.1]:   File "/app/app/api/imports.py", line 15, in import_event
2018-01-31T14:49:27.765761+00:00 app[web.1]:     file_path = get_file_from_request(['zip'])
2018-01-31T14:49:27.765762+00:00 app[web.1]:   File "/app/app/api/helpers/import_helpers.py", line 90, in get_file_from_request
2018-01-31T14:49:27.765763+00:00 app[web.1]:     raise NotFoundError(source='{}', detail='File not found')
2018-01-31T14:49:27.765764+00:00 app[web.1]: NotFoundError: <app.api.helpers.errors.NotFoundError instance at 0x7f93a1d78638>
```

### PR DESCRIPTION
**Pull request by YogeshSharma0201**
Fixes #4613, #5513

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:

Import Event api is working now and through /import-jobs api Import jobs list can be retrieved and displayed on frontend.

#### Changes proposed in this pull request:

- Event Model is fixed so that it accepts event_orga_id
- Import Jobs schema and api added.
- Added apiblueprint for import jobs api
- Import Jobs relationship are not added in the schema as it was not necessary for this PR, Let me know if that is required too.

### CODE REVIEW

**Comment by YogeshSharma0201** [comment]
@CosmicCoder96 do you think these codacy issues need to be fixed? i cannot change variable name 'id' and no methods are currently required in impor_job schema.

File: app/api/helpers/utilities.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def get_filename_from_cd(cd):
 def write_file(file, data):
     """simple write to file"""
     fp = open(file, 'w')
-    fp.write(str(data, 'utf-8'))
+    fp.write(str(data))
```
Comment (iamareebjamal) [inline app/api/helpers/utilities.py]: Why?

File: docs/api/api_blueprint.apib [hunk 2]
```diff
@@ -25517,3 +25518,63 @@ This Group's APIs can be used for adding a particular event to the favourite lis
...
+                "type": "import-job",
+                "attributes": {
+                    "result-status": "SUCCESS",
+                    "starts-at": "2019-01-08T17:24:01.971880+00:00",
+                    "task": "fcc49d97-aff8-4f77-a98b-cfcdbae9f5eb",
+                    "result": "5"
+                },
+                "id": "12",
+                "links": {
+                    "self": "/v1/import-jobs?id=12"
```
Comment (iamareebjamal) [inline docs/api/api_blueprint.apib]: This is REST API. It should be `/v1/import-jobs/12`

**Comment by CosmicCoder96** [comment]
@YogeshSharma0201 update this please

**Comment by YogeshSharma0201** [comment]
@CosmicCoder96 please review

File: app/api/__init__.py [hunk 3]
```diff
@@ -623,3 +624,7 @@
 
 # Full text search w/ Elastic Search
 api.route(EventSearchResultList, 'event_search_results', '/search/events')
+
+# Import Jobs
+api.route(ImportJobList, 'image_job_list', '/import-jobs')
```
Comment (CosmicCoder96) [inline app/api/__init__.py]: why image? Keep it `import_job_list`

File: app/api/__init__.py [hunk 4]
```diff
@@ -623,3 +624,7 @@
...
+api.route(ImportJobDetail, 'image_job_detail', '/import-jobs/<int:id>')
```
Comment (CosmicCoder96) [inline app/api/__init__.py]: `import_job_detail`

File: app/api/helpers/import_helpers.py [hunk 5]
```diff
@@ -355,10 +355,8 @@ def import_event_json(task_handle, zip_path):
         new_event = Event(**data)
         db.session.add(new_event)
         db.session.commit()
-        write_file(
-            path + '/social_links',
-            json.dumps(data.get('social_links', []))
-        )  # save social_links
+        with open(path + '/social_links', 'w') as file:
```
Comment (CosmicCoder96) [inline app/api/helpers/import_helpers.py]: why?

File: app/api/import_jobs.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+from app.models import db
+from app.models.import_job import ImportJob
+from app.api.helpers.permissions import jwt_required, is_user_itself
+
+
+class ImportJobList(ResourceList):
+    """
+    List ImportJob
+    """
+    decorators = (jwt_required, is_user_itself,)
```
Comment (CosmicCoder96) [inline app/api/import_jobs.py]: why `is_user_itself`?

File: app/api/schema/import_jobs.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    """
+    Api schema for ImportJob Model
+    """
+
+    class Meta:
+        """
+        Meta class for ImportJob Api Schema
+        """
+        type_ = 'import-job'
+        self_view = 'v1.image_job_detail'
```
Comment (CosmicCoder96) [inline app/api/schema/import_jobs.py]: `import_job_detail`

File: docs/api/api_blueprint.apib [hunk 8]
```diff
@@ -25515,3 +25516,63 @@ This Group's APIs can be used for adding a particular event to the favourite lis
...
         }
+
+
+# Group Import Jobs
+
+This Group's APIs are used for getting info of import jobs
+
+## Import Jobs Collection [/v1/import-jobs{?page%5bsize%5d,page%5bnumber%5d,sort,filter}]
+
+### List All Orders [GET]
```
Comment (CosmicCoder96) [inline docs/api/api_blueprint.apib]: List all Jobs. Not orders.

**Comment by CosmicCoder96** [comment]
@YogeshSharma0201 updates?

File: app/api/helpers/import_helpers.py (hunk 5 above)
Comment (YogeshSharma0201) [reply app/api/helpers/import_helpers.py]: @CosmicCoder96 i can't use write_file() because it tries to encode the data in utf-8 which gives errors.

Comment (iamareebjamal) [reply app/api/helpers/import_helpers.py]: What errors?

Comment (YogeshSharma0201) [reply app/api/helpers/import_helpers.py]: @iamareebjamal  `decoding str is not supported`

**Comment by YogeshSharma0201** [comment]
@CosmicCoder96 please review

**Comment by CosmicCoder96** [comment]
@iamareebjamal Can you help out with the error mentioned by @YogeshSharma0201. What could be a possible solution?

**Comment by iamareebjamal** [comment]
`write_file` takes bytes as argument and json.dumps returns string. 

Add .encode('utf-8') after json.dumps call

```
json.dumps(data.get(..., [])).encode('utf-8')
```

**Comment by CosmicCoder96** [comment]
@YogeshSharma0201 Please update this now with the fix mentioned by @iamareebjamal

**Pull request merged**